File: old_backend/app/Models/MovimientoInventario.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use App\Models\Material;
use App\Models\Mueble;
use App\Models\Venta;
use App\Models\Produccion;
use App\Models\CompraMaterial;
use App\Models\Devolucion;
use App\Models\Empleado;

class MovimientoInventario extends Model
{
    use HasFactory;

    protected $table = 'movimientos_inventario';
    protected $primaryKey = 'id_mov';
    public $timestamps = false;

    protected $fillable = [
        'cod_mov',
        'tipo_mov',
        'cantidad',
        'fecha_mov',
        'id_mat',
        'id_mue',
        'id_ven',
        'id_pro',
        'id_comp',
        'id_dev',
        'id_emp',
        'motivo',
        'stock_anterior',
        'stock_posterior',
    ];

    public function material()
    {
        return $this->belongsTo(Material::class, 'id_mat', 'id_mat');
    }
    public function mueble()
    {
        return $this->belongsTo(Mueble::class, 'id_mue', 'id_mue');
    }
    public function venta()
    {
        return $this->belongsTo(Venta::class, 'id_ven', 'id_ven');
    }
    public function produccion()
    {
        return $this->belongsTo(Produccion::class, 'id_pro', 'id_pro');
    }
    public function compraMaterial()
    {
        return $this->belongsTo(CompraMaterial::class, 'id_comp', 'id_comp');
    }
    public function devolucion()
    {
        return $this->belongsTo(Devolucion::class, 'id_dev', 'id_dev');
    }
    public function empleado()
    {
        return $this->belongsTo(Empleado::class, 'id_emp', 'id_emp');
    }


    
}

File: backend/src/main/java/com/changuitostudio/backend/domain/model/MovimientoInventario.java
package com.changuitostudio.backend.domain.model;

import java.time.LocalDateTime;

public class MovimientoInventario {
    private Long id;
    private String tipoMov;
    private LocalDateTime fechaMov;
    private Material material;
    private Mueble mueble;
    private Double cantidad;
    private Double stockAnterior;
    private Double stockPosterior;
    private Venta venta;
    private Produccion produccion;
    private CompraMaterial compra;
    private Devolucion devolucion;
    private String motivo;
    private Empleado empleado;
    private String codMov;

    public MovimientoInventario() {
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }
    public String getTipoMov() {
        return tipoMov;
    }

    public void setTipoMov(String tipoMov) {
        this.tipoMov = tipoMov;
    }
    public LocalDateTime getFechaMov() {
        return fechaMov;
    }

    public void setFechaMov(LocalDateTime fechaMov) {
        this.fechaMov = fechaMov;
    }
    public Material getMaterial() {
        return material;
    }

    public void setMaterial(Material material) {
        this.material = material;
    }
    public Mueble getMueble() {
        return mueble;
    }

    public void setMueble(Mueble mueble) {
        this.mueble = mueble;
    }
    public Double getCantidad() {
        return cantidad;
    }

    public void setCantidad(Double cantidad) {
        this.cantidad = cantidad;
    }
    public Double getStockAnterior() {
        return stockAnterior;
    }

    public void setStockAnterior(Double stockAnterior) {
        this.stockAnterior = stockAnterior;
    }
    public Double getStockPosterior() {
        return stockPosterior;
    }

    public void setStockPosterior(Double stockPosterior) {
        this.stockPosterior = stockPosterior;
    }
    public Venta getVenta() {
        return venta;
    }

    public void setVenta(Venta venta) {
        this.venta = venta;
    }
    public Produccion getProduccion() {
        return produccion;
    }

    public void setProduccion(Produccion produccion) {
        this.produccion = produccion;
    }
    public CompraMaterial getCompra() {
        return compra;
    }

    public void setCompra(CompraMaterial compra) {
        this.compra = compra;
    }
    public Devolucion getDevolucion() {
        return devolucion;
    }

    public void setDevolucion(Devolucion devolucion) {
        this.devolucion = devolucion;
    }
    public String getMotivo() {
        return motivo;
    }

    public void setMotivo(String motivo) {
        this.motivo = motivo;
    }
    public Empleado getEmpleado() {
        return empleado;
    }

    public void setEmpleado(Empleado empleado) {
        this.empleado = empleado;
    }
    public String getCodMov() {
        return codMov;
    }

    public void setCodMov(String codMov) {
        this.codMov = codMov;
    }
}
